### ISSUE TITLE
If BokehJS isn't available, error message should be displayed

### ISSUE THREAD

**Issue by gkennos** [bokeh/bokeh#9684]
I'm trying to use Bokeh behind a proxy that doesn't allow outbound connections to https://cdn.pydata.org, which means that BokehJS can't be downloaded.  I can't do bokeh.resources = inline, as I have been unable to install bokehjs and its dependencies using npm due to the same lack of connectivity.

When I try to display a figure, nothing happens, but when I inspect the element, I can see the root cause is that BokehJS is not available (screenshot below).  I am getting pydata.org whitelisted, as we are not able to whitelist github and therefore it is challenging to get npm dependencies to install properly for local installation.  I think this should resolve the issue, but is it possible to have an error message at this point instead of just not rendering the script.

Alternative would be to have jupyter-bokeh include the installation of bokehjs locally, but I don't know the rationale behind having it loaded from the CDN, so assume it doesn't do that for a specific reason.

OS: Amazon Linux 
Browser: Firefox 71.0
Python: 3.6.8
Jupyter lab version: 1.2.4
Bokeh: 1.3.4
jupyter-bokeh: 1.1.1

![Screen Shot 2019-12-20 at 12 45 29 pm](https://user-images.githubusercontent.com/12161195/71223013-a60fbc80-2326-11ea-956d-7310bc4f9470.png)

**Comment by bryevdv** [comment]
I've transferred this issue to the  main repo. The actual place that would need to be updated is here in ` core/_templates/autoload_nb_js.js` and he other repo is only worked on sporadically so it's much more likely get ignored there.

**Comment by bryevdv** [comment]
The relevant code in the template is here:
```  function display_loaded() {
    var el = document.getElementById({{ elementid|json }});
    if (el != null) {
      el.textContent = "BokehJS is loading...";
    }
    if (root.Bokeh !== undefined) {
      if (el != null) {
        el.textContent = "BokehJS " + root.Bokeh.version + " successfully loaded.";
      }
    } else if (Date.now() < root._bokeh_timeout) {
      setTimeout(display_loaded, 100)
    }
```
I think a final `else` block can just be added to add the failure message.

**Comment by ayushiitr2901** [comment]
I think the issue here is that the 'loading failed' text message is not shown if the BokehJS is unable to load(might be because of some disturbance in connectivity).
As suggested above by @bryevdv, I will go ahead and add an else statement at the end of 'display_loaded' function that changes the value of 'el' variable to "Failed to load BokehJS..."

**Comment by arihans** [comment]
@bryevdv what is `_bokeh_timeout` ? I searched for it but didn't found any declaration .

```
$ grep -rnw './bokeh' -e '_bokeh_timeout'

./bokeh/core/_templates/autoload_nb_js.js:134:  if (typeof (root._bokeh_timeout) === "undefined" || force === true) {
./bokeh/core/_templates/autoload_nb_js.js:135:    root._bokeh_timeout = Date.now() + {{ timeout|default(0)|json }};
./bokeh/core/_templates/autoload_nb_js.js:164:    } else if (Date.now() < root._bokeh_timeout) {
./bokeh/core/_templates/autoload_nb_js.js:178:    } else if (Date.now() < root._bokeh_timeout) {
```

**Comment by arihans** [comment]
@bryevdv is this open to work.

**Comment by mattpap** [comment]
Error reporting was implemented in PR #13523.
